### DISCUSSION TITLE
redis缩容后redisson客户端报错 unable to write command into connection! Increase connection pool size

### DISCUSSION THREAD

**Discussion by ice-llz** [redisson/redisson#6059] [Q&A]
项目里使用的版本号为3.17.5
线上redis服务器缩容（减少了分片）后，代码里用了redisson里的 tryLock 方法，报出如上错误。redis监控显示在缩容后 连接数立马飙升了4倍左右，有大神清楚这是为什么吗？重启服务恢复

**Comment by ice-llz** [comment]
> 项目里使用的版本号为3.17.5
> 线上redis服务器缩容（减少了分片）后，代码里用了redisson里的 tryLock 方法，报出如上错误。redis监控显示在缩容后 连接数立马飙升了4倍左右，有大神清楚这是为什么吗？重启服务恢复

无慢查询，无大key，所以应该不是redis服务处理慢导致链接数不够

**Reply by mrniko** [reply]
Can you try 3.33.0 version? Several issues with connection leak have been fixed since 3.17.5.

**Reply by ice-llz** [reply]
> Can you try 3.33.0 version? Several issues with connection leak have been fixed since 3.17.5.

yes I can use the newest version. but I want to know the reason if anybody could tell me. thanks
